src/main/java/org/checkerframework/specimin/InheritancePreserveVisitor.java

```diff
@@ -49,17 +50,47 @@ public void emptyAddedClasses() {
   public Visitable visit(ClassOrInterfaceDeclaration decl, Void p) {
     if (usedClass.contains(decl.resolve().getQualifiedName())) {
       for (ClassOrInterfaceType extendedType : decl.getExtendedTypes()) {
-        String qualifiedName;
         try {
-          qualifiedName = extendedType.resolve().getQualifiedName();
+          updateAddedClassWithQualifiedClassName(extendedType.resolve().getQualifiedName());
+          if (extendedType.getTypeArguments().isPresent()) {
+            for (Type typeAgrument : extendedType.getTypeArguments().get()) {
+              updateAddedClassWithQualifiedClassName(typeAgrument.resolve().describe());
+            }
+          }
         }
         // since Specimin does not create synthetic inheritance for interfaces.
         catch (UnsolvedSymbolException | UnsupportedOperationException e) {
           continue;
         }
-        addedClasses.add(qualifiedName);
       }
     }
     return super.visit(decl, p);
   }
+
+  /**
+   * (This method is copied from TargetMethodFinderVisitor).
```

Comment:
Rather than copying, we should create a shared static utility method.

Reply:
(for now, it could be in either visitor, though at some point we'll want to consolidate these kind of utility methods in one place)